src/content/reference/eslint-plugin-react-hooks/lints/set-state-in-render.md

```diff
@@ -0,0 +1,92 @@
+---
+title: set-state-in-render
+---
+
+<Intro>
+
+Validates against setting state during render, which can trigger additional renders and potential infinite render loops.
+
+</Intro>
+
+## Rule Details {/*rule-details*/}
+
+Calling `setState` during render triggers another render before the current one finishes. This creates an infinite loop that crashes your app.
+
+## Common Violations {/*common-violations*/}
+
+### Invalid {/*invalid*/}
+
+```js {expectedErrors: {'react-compiler': [4]}}
+// ❌ setState directly in render
+function Component({value}) {
+  const [count, setCount] = useState(0);
+  setCount(value); // Infinite loop!
+  return <div>{count}</div>;
+}
+```
+
+### Valid {/*valid*/}
+
+```js
+// ✅ Derive during render
+function Component({items}) {
+  const sorted = [...items].sort(); // Just calculate it in render
+  return <ul>{sorted.map(/*...*/)}</ul>;
+}
+
+// ✅ Set state in event handler
+function Component() {
+  const [count, setCount] = useState(0);
+  return (
+    <button onClick={() => setCount(count + 1)}>
+      {count}
+    </button>
+  );
+}
+
+// ✅ Derive from props instead of setting state
+function Component({user}) {
+  const name = user?.name || '';
+  const email = user?.email || '';
+  return <div>{name}</div>;
+}
+```
+
+## Troubleshooting {/*troubleshooting*/}
+
+### I need to reset state when props change {/*reset-state-on-prop-change*/}
+
+Sometimes you want to reset internal state when a prop changes:
+
+```js
+// ❌ Wrong approach
+function EditForm({itemId}) {
+  const [formData, setFormData] = useState({});
+
+  // This creates infinite renders
+  if (itemId !== lastItemId) {
+    setFormData({});
+  }
+
+  return <form>...</form>;
```

Comment:
This is valid, but the pattern would involve storing lastItemId in another state and also resetting that state value in the conditional

Reply:
let me change up this example, it's not very good